### ISSUE TITLE
approx_constant "Known problems" section is incorrect

### ISSUE THREAD

**Issue by kraai** [rust-lang/rust-clippy#4025]
The documentation for `approx_constant` says in the "Known problems" section:

> If you happen to have a value that is within 1/8192 of a known constant, but is not and should not be the same, this lint will report your value anyway. We have not yet noticed any false positives in code we tested clippy with (this includes servo), but YMMV.

The function that compares the values has the following header comment:

```
/// Returns `false` if the number of significant figures in `value` are                                                                               
/// less than `min_digits`; otherwise, returns true if `value` is equal                                                                               
/// to `constant`, rounded to the number of digits present in `value`.                                                                                
```

There's nothing in that comment, or that I can see in the code, referring to 1/8192, so I think the documentation is incorrect.

**Comment by flip1995** [comment]
I don't really know where the number `1/8192`(=`2^-13`) is coming from. I assume it comes from the precision of float numbers expressed in decimal digits. 

cc @llogiq wrote the lint and documentation ~4 years ago. Maybe he can shed some light onto this magic number.

**Comment by flip1995** [comment]
In the initial implementation there was a magic `EPSILON_DIVISOR` constant: https://github.com/rust-lang/rust-clippy/blob/a927bc0b7051b798dff0e00c8f05875bedab90de/src/approx_const.rs#L23 (old like from 09/2015)

This constant was removed in #408 and I think this `Known Problems` is not an issue anymore.
